docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,489 @@
+     cmake --build .
+     cmake --install . --prefix <path_to_install_dir>                    # required to have full package structure
+
+In the above, the ``$CXX_COMPILER`` and ``$C_COMPILER`` should be set to ``icpx`` and ``icx`` respectively when using the Intel(R) oneAPI DPC++ Compiler,
+or ``clang++`` and ``clang`` respectively when using the Open DPC++ Compiler. 
+
+Backends should be enabled by setting ``-DENABLE_<BACKEND_NAME>_BACKEND=ON`` for each desired backend. 
+By default, only the ``MKLGPU`` and ``MKLCPU`` backends are enabled.
+Multiple backends for multiple device vendors can be enabled at once (albeit with limitations when using portBLAS and portFFT).
+For examples, see the section `CMake invocation examples`_.
+The supported backends for the compilers are given in the table at
+`oneMKL supported configurations table <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#supported-configurations>`_, 
+and the CMake option names are given in the table below.
+Some backends may require additional parameters to be set. See the relevant section below for additional guidance.
+
+If a backend library supports multiple domains (i.e. BLAS, LAPACK, DFT, RNG), it may be desirable to only enable selected domains.
+For this, the ``TARGET_DOMAINS`` variable should be set.
+See the section `TARGET_DOMAINS`_.
+
+By default, the library also additionally builds examples and tests.
+These can be disabled by setting the parameters ``BUILD_FUNCTIONAL_TESTS`` and ``BUILD_EXAMPLES`` to off.
+Building the functional tests may require additional external libraries.
+See the section `Building and running tests`_ for more information.
+
+The most important supported build options are:
+
+.. list-table::
+   :header-rows: 1
+
+   * - CMake Option
+     - Supported Values
+     - Default Value 
+   * - ENABLE_MKLCPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_MKLGPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_CUBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUFFT_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CURAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_NETLIB_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCFFT_BACKEND
+     - True, False
+     - False    
+   * - ENABLE_ROCSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCRAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_MKLCPU_THREAD_TBB
+     - True, False
+     - True      
+   * - ENABLE_PORTBLAS_BACKEND
+     - True, False
+     - False      
+   * - ENABLE_PORTFFT_BACKEND
+     - True, False
+     - False      
+   * - BUILD_FUNCTIONAL_TESTS
+     - True, False
+     - True      
+   * - BUILD_EXAMPLES
+     - True, False
+     - True      
+   * - TARGET_DOMAINS (list)
+     - blas, lapack, rng, dft
```

Comment:
`sparse_blas` could be added to the list here for completeness. It is added to the default list of domains and mentioned in the README as there is some support for it with MKLCPU and MKLGPU. Similarly in the `TARGET_DOMAINS` section below.

Reply:
Added in [90ce137](https://github.com/oneapi-src/oneMKL/pull/510/commits/90ce13728ac9a2a6a4ed4f969ca1ccb9de50e34b). Additionally, I realised that I should have included all the domains in the AdaptiveCpp notes since it supports MKLCPU.